File: week1/cash.c
#include <cs50.h> //includes all libraries
#include <stdio.h>
#include <math.h>

int coins = 0; //sets coins to 0
int q = 0; //sets q, d, n ,p all to 0 for test counts
int d = 0;
int n = 0;
int p = 0;

int cashier(int cents); //calls fxn cashier
int change(void); //calls fxn change

int main(void) //defines main fxn
{
    int cents = change(); // creates and sets int cents  by calling change fxn

    if (cents <= 0) //if < 0 prints error message and the calls change fxn again
    {
        printf("please enter a postive number\n");
        cents = change();
    }

    while (cents >= 1) // if equal to or greater than 1
    {
        cents = cashier(cents); //runs cashier fxn
    }


    //printf("Q: %i\n", q);
    //printf("D: %i\n", d);
    //printf("N: %i\n", n);
    //printf("P: %i\n", p);
    printf("%i\n", coins); //after cashier fxn runs prints the total coin count

}

int change(void) //defines the change fxn with no input and an int output
{
    float x = get_float("Change Owed: "); // gets user to input a float for x
    float y = x * 100; // changes x from dollars to cents and assigns to y
    float cents = round(y); // creates float cents of a rounded y

    return cents; //returns cents
}

int cashier(int cents) //defines the cashier fxn
{
    if (cents >= 100) // if cent greater than or eqaul to 1 dollar
    {
        cents = cents - 100;
        coins += 4; // adds 4 quaters and coins to count
        q += 4;
        //printf("qcoins %i\n", coins);
        //printf("qq %i\n", q);
    }
    if (cents >= 25) // if over 25 cents takes a q away from cents and adds to coin count
    {
        cents = cents - 25;
        coins++;
        q++;
        return cents;
    }

    if (cents >= 10) //same for dimes or 10 cents
    {
        cents = cents - 10;
        coins++;
        d++;
        return cents;
    }

    if (cents >= 5) //same for nickles
    {
        cents = cents - 5;
        coins++;
        n++;
        return cents;
    }

    if (cents >= 1) // same for pennys
    {
        cents = cents - 1;
        coins++;
        p++;
        return cents;
    }
    return cents; //returns cents to main fxn


}

File: week6/cash.py
from cs50 import get_float


def main():
    # defines the main fxn
    # sets var coins to 0; can py have gloabal vars?
    coins = 0
    cents = change()
    # need to call cashier fxn
    if cents >= 1:
        while cents >= 1:
            # set cents = to the out put of cashier
            # add to the coin counter
            coins += 1
            cents = cashier(cents)
            # also get the new coin value from cashier fxn
            if cents == 0:
                break

    else:
        return change()

    print(coins)


def change():
    # x is the float input provided by user
    x = get_float("Change owed: ")
    # y is x multiplied by 100 to change it from dollars to cents
    y = x * 100
    # cents is the var that is rounded y
    cents = round(y)
    return cents


def cashier(cents):

    if cents > 24:
        cents = cents - 25
        return cents

    elif cents > 9:
        cents = cents - 10
        return cents

    elif cents > 4:
        cents = cents - 5
        return cents

    elif cents > 0:
        cents = cents - 1
        return cents
    else:
        return cents


main()
